tests/TestTCP.hs

```diff
@@ -694,15 +694,14 @@ testReconnect = do
     -- This might time out or not, depending on whether the server closes the
     -- socket before or after we can send the RequestConnectionId request
     resultConnect <- timeout 500000 $ connect endpoint theirAddr ReliableOrdered defaultConnectHints
+    WaitSocketFree wait <- takeMVar socketClosed
+    wait
     case resultConnect of
       Nothing -> return ()
       Just (Left (TransportError ConnectFailed _)) -> return ()
       Just (Left err) -> throwIO err
       Just (Right _) -> throwIO $ userError "testConnect: unexpected connect success"
 
-    WaitSocketFree wait <- takeMVar socketClosed
-    wait
-
     -- Second attempt: server accepts but then closes the socket. We expect
     -- either a failed connection, or a subsequent send to fail.
     resultConnect <- timeout 500000 $ connect endpoint theirAddr ReliableOrdered defaultConnectHints
```

Comment:
If the timeout occurs before the server gets the connection request, the test would block. I don't understand what this timeout call accomplishes.